In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats
data = pd.read_csv('C:/Users/Pyter/DA/BaiTap/GDPlist.csv',header=0 , encoding = "ISO-8859-1")
data

Country      Continent  GDP (millions of US$)
0     Afghanistan           Asia                  18181
1         Albania         Europe                  12847
2         Algeria         Africa                 190709
3          Angola         Africa                 100948
4       Argentina  South America                 447644
..            ...            ...                    ...
120    Uzbekistan           Asia                  45353
121     Venezuela  South America                 315841
122       Vietnam           Asia                 122722
123         Yemen         Africa                  33675
124        Zambia         Africa                  19206

[125 rows x 3 columns]

Câu 1:\
Trung bình GDP của các quốc gia trên thế giới là 500 tỉ usd/năm

In [2]:
t_statistic, p_value = stats.ttest_1samp(data['GDP (millions of US$)'], 500000)
print('t_statistic = ',t_statistic)
print('p_value = ',p_value)

t_statistic =  0.3757438669518338
p_value =  0.7077493494055246


Kiểm định giả thuyết:\
Với p_value =  0.7077493494055246 > alpha = 0.05 nên ta không thể kết luận rằng trung bình GDP của các quốc gia trên thế giới khác 500 tỷ USD/năm.

Câu 2:\
GDP trung bình ở các quốc gia ở châu Âu cao hơn châu Á không

In [3]:
df = data[data['Continent']=='Europe']
df1 = data[data['Continent']=='Asia']

In [5]:
t_statistic, p_value = stats.ttest_ind(df['GDP (millions of US$)'],df1['GDP (millions of US$)'], equal_var=True)
print('t_statistic = ',t_statistic)
print('p_value = ',p_value)

t_statistic =  -0.5586877419842815
p_value =  0.5781326801926459


Kiểm định giả thuyết:
Với p_value =  0.5781326801926459 > alpha = 0.05 với kết quả này, ta không thể kết luận rằng GDP trung bình ở các quốc gia châu Âu cao hơn châu Á.

Câu 3:\
GDP trung bình của các quốc gia ở châu Âu và châu Mỹ là bằng nhau

In [6]:
df2 = data[(data['Continent']=='South America') | (data['Continent']=='North America')]

In [7]:
t_statistic, p_value = stats.ttest_ind(df['GDP (millions of US$)'],df2['GDP (millions of US$)'], equal_var=True)
print('t_statistic = ',t_statistic)
print('p_value = ',p_value)

t_statistic =  -1.0372877375171978
p_value =  0.3041408163690613


Kiểm định giả thuyết:\
Với p_value =  0.3041408163690613 < alpha = 0.05 với kết quả này, ta không thể kết luận rằng GDP trung bình của các quốc gia ở châu Âu và châu Mỹ khác nhau.